macros/calendar_date/string_to_date.sql

```diff
@@ -0,0 +1,54 @@
+{# 
+    currently supports dates stored like 'Apr 1, 2022'
+    other formats to be covered: 
+    - 'Apr 01, 2022' -> 'MMM D, YYYY'
+    - 'April 1, 2022' -> 'Month D, YYYY'
+    - 'April 01, 2022' -> 'Month DD, YYYY'
+    - '1 April 2022' -> 'D Month YYYY'
+    - '1 Apr 2022' -> 'D MMM YYYY'
+    - '4/1/22' -> 'M/D/YY'
+    - '4/1/2022' -> 'M/D/YYYY'
+    - '04/01/22' -> 'MM/DD/YY'
+    - '04/01/2022' -> 'MM/DD/YYYY'
+
+    If we want to support Day-Month-Year formats:
+    - '1/4/22' -> 'D/M/YYY'
+    - '1/4/2022' -> 'D/M/YYYY'
+    - '01/04/22' -> 'DD/MM/YY'
+    - '01/04/2022' -> 'DD/MM/YYYY'
+
+    Anything else?
+#}
+
+{%- macro string_to_date(date_string, format='MMM D, YYYY') -%}
```

Comment:
it doesn't look like the `format` parameter is used anywhere within `string_to_date`?

Reply:
Also, this should probably be named `date_format`.